### PR TITLE
Add support for adis1646x and adis1647x families

### PR DESCRIPTION
**Pull request by rbolboac**
## Pull Request Description

Add support for adis1646x and adis1647x families

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: doc/sphinx/source/projects/eval-adis1657x.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+.. include:: ../../../../drivers/projects/eval-adis1657x/README.rst
```
Comment (amiclaus) [inline doc/sphinx/source/projects/eval-adis1657x.rst:1]: these paths all look strange.

File: doc/sphinx/source/projects_doc.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+*******************
+
+The following sections contain code documentation for ADI no-OS projects.
+
+INERTIAL MEASUREMENT UNITS
+==========================
+.. toctree::
+   :maxdepth: 1
+
+   ../projects/eval-adis1646x
```
Comment (amiclaus) [inline doc/sphinx/source/projects_doc.rst:12]: is `../` really needed here? wondering if we can use the same approach as the one we used for drivers.

**Review by amiclaus — COMMENTED** [review]
also, there are a lot of sphinx warnings in the documentation build.

File: doc/sphinx/source/projects/eval-adis1657x.rst (hunk 1 above)
Comment (rbolboac) [reply doc/sphinx/source/projects/eval-adis1657x.rst]: oops, typo there, updated the paths

File: doc/sphinx/source/projects_doc.rst (hunk 2 above)
Comment (rbolboac) [reply doc/sphinx/source/projects_doc.rst]: nope, it is not needed, I made the updat

**Comment by rbolboac** [comment]
> also, there are a lot of sphinx warnings in the documentation build.

thanks, I made some updates, I'll wait for the build and see if there are any other warnings introduced by me

**Comment by amiclaus** [comment]
> > also, there are a lot of sphinx warnings in the documentation build.
> 
> thanks, I made some updates, I'll wait for the build and see if there are any other warnings introduced by me

FYI, I added #1958 . after it gets merged the sphinx warnings will result in a build fail.

**Comment by rbolboac** [comment]
- master rebase
- rename dummy project to basic project

**Review by CiprianRegus — APPROVED** [review]

File: doc/sphinx/source/index.rst [hunk 3]
```diff
@@ -13,7 +13,7 @@ according to users’ requirements.
 
    We are in the process of moving the no-OS documentation from `wiki.analog.com <https://wiki.analog.com/resources/no-os>`_
    to  `github.com/analogdevicesinc/no-OS <https://github.com/analogdevicesinc/no-OS>`_.
-   
+
```
Comment (amiclaus) [inline doc/sphinx/source/index.rst:16]: this seems unrelated.

**Review by amiclaus — APPROVED** [review]

File: doc/sphinx/source/index.rst (hunk 3 above)
Comment (rbolboac) [reply doc/sphinx/source/index.rst]: thanks, removed it

**Comment by rbolboac** [comment]
- master rebase + resolve conflicts

File: doc/sphinx/source/index.rst [hunk 4]
```diff
@@ -13,7 +13,6 @@ according to users’ requirements.
 
    We are in the process of moving the no-OS documentation from `wiki.analog.com <https://wiki.analog.com/resources/no-os>`_
    to  `github.com/analogdevicesinc/no-OS <https://github.com/analogdevicesinc/no-OS>`_.
```
Comment (amiclaus) [inline doc/sphinx/source/index.rst:15]: this still seems to be unrelated change.
